contributing/topics/guide-breaking-changes.md

```diff
@@ -1,6 +1,252 @@
+    
+    func (r ResourceWithReplacement) DeprecatedInFavourOfResource() string {
+        return "azurerm_new_resource"
+    }
+    
+    ```
+
+    For Untyped Resources
+    ```go
+    func resourceExample() *pluginsdk.Resource {
+        return &pluginsdk.Resource{
+            Create: resourceExampleCreate,
+            Read:   resourceExampleRead,
+            Update: resourceExampleUpdate,
+            Delete: resourceExampleDelete,
+            
+            Timeouts: &pluginsdk.ResourceTimeout{
+            Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+            Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+            Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+            Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+            },
+            
+            DeprecationMessage: "The `azurerm_example` resource has been deprecated and will be removed in v5.0 of the AzureRM Provider"
+            ...
+        }
+    }
+    ```
+
+2. Conditionally register the resource in the `registration.go` file of the service package.
+    
+   For Typed Resources
+    ```go
+    func (r Registration) Resources() []sdk.Resource {
+        resources := []sdk.Resource{
+            MySqlFlexibleServerResource{},
+        }
+        
+        if !features.FivePointOhBeta() {
+            resources = append(resources, ExampleResource{})
+        }
+        
+        return resources
+    }
+    ```
+    
+    For Untyped Resources
+    ```go
+    func (r Registration) SupportedResources() map[string]*pluginsdk.Resource {
+        resources := map[string]*pluginsdk.Resource{
+            "azurerm_mysql_flexible_server": resourceMysqlFlexibleServer(),
+    
+        }
+    
+        if !features.FivePointOhBeta() {
+            resources["azurerm_example"] = resourceExample()
+        }
+    
+        return resources
+    }
+    ```
+
+3. Skip all tests related to the deprecated resource.
+
+    ```go
+    func TestAccExample_basic(t *testing.T) {
+        if features.FivePointOhBeta() {
+            t.Skipf("Skipping since `azurerm_example` is deprecated and will be removed in 5.0")
+        }
+        data := acceptance.BuildTestData(t, "azurerm_example", "test")
+        r := ExampleResource{}
+        
+        data.ResourceTest(t, r, []acceptance.TestStep{
+        {
+            Config: r.basic(data),
+            Check: acceptance.ComposeTestCheckFunc(
+                check.That(data.ResourceName).ExistsInAzure(r),
+            ),
+        },
+            data.ImportStep()
+        })
+    }
+    ```
+
+4. Update the upgrade guide under `website/docs/5.0-upgrade-guide.markdown`.
+
+   ```markdown
+   ## Removed Resources
+   
+   ### `azurerm_example`
+   
+   This deprecated resources has been removed from the Azure Provider.
+   ```
+   **Note:** The Resource documentation does not need to be updated with a note since the provider will inform any users using the resource of the deprecation with a warning.
```

Comment:
I wonder if it is worth throwing a note into the resource docs anyway?

I'm thinking of a use case where I'm writing out a resource and get it all written just to find out when running Terraform that there is a new one to use instead and I could have written it out the new way instead of the old.